packages/client/src/pages/home/map.tsx

```diff
@@ -84,9 +85,12 @@ const Map = () => {
     lat: location.lat,
     lng: location.lon,
   });
+
   const [name, setName] = React.useState(location.name);
 
   if (initialLocation === undefined) {
+    // TODO: this block is causing getPostsByFilter to be called twice when the page is refreshed
+    // Another issue: user location is fetched here AND in post-slice
```

Comment:
Assuming this code block only needs to run once, to address these two issues maybe we could:
- Get rid of the `location` slice so the current location is only maintained in one place (i.e. in the post-slice). 
- Implement an action in the post-slice called `initLocation` that does sth like
```
navigator.geolocation.getCurrentPosition(function (position) {
  state.locationFilter = {
    lat: position.coords.latitude,
    lon: position.coords.longitude,
    name: undefined,
  };
});
```
- Dispatch `initLocation` in a `useEffect` hook in `HomePage` or `Map` (to avoid redundant calls/renders)
- Or we could just call the above code block in a `useEffect` hook and set it in the store (no additional action necessary)

Reply:
I did it the second way and it fixes most of the problem

Just noticed that we also called getPostsByFilter for the initialLocation, but at least the duplicate requests for the detected location are gone :monkey_face: :monkey_face: